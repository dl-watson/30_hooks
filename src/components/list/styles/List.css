.List {
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    list-style-type: none;

    .listItem {
        display: flex;
        flex-direction: column-reverse;
        align-items: center;
        margin: 20px;
    }

    img {
        width: 20vh;
    }
}