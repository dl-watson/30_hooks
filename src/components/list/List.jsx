import React from "react";
import { Link } from "react-router-dom";
import PropTypes from "prop-types";
import uuid from "react-uuid";
import styles from "./styles/List.css";

const List = ({ characters }) => {
  const characterList = characters.map((character) => {
    return (
      <li key={uuid()} className={styles.listItem}>
        {character.name}
        <Link to={`/details/${character.name}`}>
          <img src={character.image} />
        </Link>
      </li>
    );
  });

  return (
    <ul data-testid="unordered-list" className={styles.List}>
      {characterList}
    </ul>
  );
};

List.propTypes = {
  characters: PropTypes.array,
};

export default List;
