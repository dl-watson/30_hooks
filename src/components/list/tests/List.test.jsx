import React from "react";
import { BrowserRouter as Router } from "react-router-dom";
import { render, cleanup } from "@testing-library/react";
import List from "../List";

describe("List component", () => {
  afterEach(() => cleanup());
  it("renders List", () => {
    const { asFragment } = render(
      <Router>
        <List characters={[{}]} />
      </Router>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
