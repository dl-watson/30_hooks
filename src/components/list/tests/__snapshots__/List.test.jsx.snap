// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component renders List 1`] = `
<DocumentFragment>
  <ul
    class="List"
    data-testid="unordered-list"
  >
    <li
      class="listItem"
    >
      <a
        href="/details/undefined"
      >
        <img />
      </a>
    </li>
  </ul>
</DocumentFragment>
`;
